### PR TITLE
polyfilled Object.assign for IE

### LINKED ISSUE #60: Tests failing in IE, can't count on Object.assign existing
e.g. https://travis-ci.org/PolymerElements/carbon-route/builds/119073713

### PR DESCRIPTION
**Pull request by e111077**
fixes #60. window.location.origin in [iron-location](https://github.com/PolymerElements/iron-location/blob/db8f5068ff8a91fa97edc80b41a9a9338ee84a1f/iron-location.html#L284) needs to be polyfilled for IE so that these tests will pass. [This PR](https://github.com/PolymerElements/iron-location/pull/24) should fix this issue.

### CODE REVIEW

File: carbon-route.html [hunk 1]
```diff
@@ -161,6 +161,23 @@
       this.linkPaths('tail.__queryParams', 'route.__queryParams');
     },
 
+    // IE Object.assign polyfill
+    __assign: function(target) {
+      target = Object(target);
+
+      for (var i = 1; i < arguments.length; i++) {
```
Comment (rictic) [inline carbon-route.html]: `arguments` can be hard for some JS engines to do quickly, and we're only ever calling this with 2 args, so it's probably worth it to just make source an explicit parameter and drop the use of `arguments`

File: carbon-route.html [hunk 2]
```diff
@@ -161,6 +161,23 @@
       this.linkPaths('tail.__queryParams', 'route.__queryParams');
     },
 
+    // IE Object.assign polyfill
+    __assign: function(target) {
+      target = Object(target);
```
Comment (rictic) [inline carbon-route.html]: This line is unnecessary for our use case

**Comment by rictic** [comment]
Please rebase to get the `.travis.yml` file

**Comment by e111077** [comment]
@rictic Did the things you asked, but for some reason, it times out on travis but not locally on Chrome :/

**Comment by rictic** [comment]
This looks good, a definite improvement, but we're getting a failure in IE 10 it looks like.

**Comment by rictic** [comment]
We should merge this though, as it's a good fix in itself. Filed the IE 10 bug as #65

**Comment by rictic** [comment]
LGTM.

File: carbon-route.html [hunk 3]
```diff
@@ -171,6 +171,17 @@
       this.linkPaths('tail.__queryParams', 'route.__queryParams');
     },
 
+    // IE Object.assign polyfill
+    __assign: function(target, source) {
+      if (source !== undefined || source !== null) {
```
Comment (rictic) [inline carbon-route.html]: `if (source != null) {`

As written this is always true

Comment (e111077) [inline carbon-route.html]: Logic sometimes fails me :/ fixed!

**Comment by rictic** [comment]
LGTM!

**Pull request merged**